### ISSUE TITLE
Scrutinise the budgets

### ISSUE THREAD

**Issue by aubergine10** [dymanoid/RealTime#60]
Most budgets aren't affected by sun position. That situation was ignorable in the vanilla game, but becomes somewhat cringeworthy when RealTime is applied because the city starts behaving as it should (eg. people are at work when they should be at work, which is unaffected by sun position, rather than a random mess like vanilla).

For reference (probably missing stuff - eg. there's a heating budget, right?):

![region capture 10](https://user-images.githubusercontent.com/1386719/42642045-d7ccee6e-85ed-11e8-941f-d7fc875bf235.png)

## Electricity

Concept of budget here is meaningless IMO. If you need more power, build more power gen buildings. Increasing budget isn't going to make a wind turbine kick out more juice.

Consumption based on clock time (ie. activity of citizens) and temperature rather than just sun position, eg:

![hourly_big](https://user-images.githubusercontent.com/1386719/42642391-af914ba6-85ee-11e8-947e-ca15e52928e4.png)

IMO separate mod should refactor this slider in to a buy/sell price for trading power with off-map cities.

## Water / Sewage

Uhm, same sort of thing as Electricity!

## Garbage

Not affected by sun position.

If anything should be main shift vs. other times.

## Healthcare and deathcare

Pretty much 24/7 these days?

Certainly nothing to do with sun position.

## Fire

Should just be single slider, fires don't care what time of day it is.

## Police

Not affected by sun position - if anything it's affected by work shifts and events? So probably peak vs. off-peak, with the mod working out from other settings when those periods are?

## Education

Not affected by sun position.

Should just be single slider to set budget during operational hours?

## Lanscaping

Relates to trees and paths.

Mostly done during daylight hours, pretty much nil outside main shift hours.

Should just be single slider defining budget during operational hours.

## Monuments and Uniques

Not affected by sun position.

Single slider defining budget during operational hours (being main and evening work shift)?

## Roads

I assume this means physical transport infrastructure (rail tracks, cable car wires, etc) but not stations or vehicles?

Again, not really affected by sun position. It's more about work hours (main workshift vs. other shifts).

## Various PT types

Definitely not affected by sun position.

IMO bulk of discussion taking place in #41 is trying to determine the 'busy' period for PT.

While there are rush hours, as we've discussed elsewhere (#31 for example) they all sort of blend in to one another. Hence the discussion in #41 to define a period (that I call "shopping hours") during which the most activity occurs. Outside that period PT provision would generally decrease.

# In Summary...

1. Sunset and sunrise are pretty much meaningless concepts in terms of budgets.
2. In vanilla game, that didn't matter because it was just complete chaos
3. RT fixes that chaos, and now I cringe every time I think of day vs. night budgets
4. Only Landscaping budget is conceivably dependant on sun position when RT is active
4. PT budgets in particular are unusable (conceptually broken) when RT is active (even dynamic sunrise/sunset feature is disabled)

**Comment by dymanoid** [comment]
See my proposal in #41. We decouple the day/night terms from the sun position. Day/night budgets make sense, because they also affect workplaces count in particular buildings and change the upkeep costs, if I recall it right.

I doubt that it makes sense to do anything with budgets in scope of **Real Time**, however... Just don't touch the sliders, if you don't like them.

**Comment by Sipke82** [comment]
Considering that #41 is completed, isn't this request completed already? Daytime (default: 6am-10pm) is now linked to budgets if i read correctly? (and is not depended on the sun). or am i missing something?

I havn't tested yet, but i guess we can now set PT budget at sleeptime to a low amount? and during daytime (6am - 10 pm) to a high amount.

**Comment by aubergine10** [comment]
Yes, that seems to be the case. For me this solution resolves the PT issues of budgeting. I guess the other types of budget are probably better suited to separate mods.

**Comment by dymanoid** [comment]
Do we close this?

**Comment by aubergine10** [comment]
yup, closing for now
